### PR TITLE
Initial commit of invocation chunking support

### PR DESCRIPTION
**Pull request by autodidaddict**
Signed-off-by: Kevin Hoffman <autodidaddict@users.noreply.github.com>

### CODE REVIEW

**Comment by autodidaddict** [comment]
Especially interested in your thoughts @stevelr as this has direct impact on `wasmbus_rpc`

File: host_core/native/hostcore_wasmcloud_native/src/lib.rs [hunk 1]
```diff
@@ -17,10 +21,17 @@ use wascap::prelude::*;
...
+mod objstore;
 mod oci;
 mod par;
 mod task;
 
+lazy_static! {
+    static ref CHUNKING_STORE: RwLock<Option<ObjectStore>> = RwLock::new(None);
+}
+
+const CHONKY_THRESHOLD_BYTES: usize = 1024 * 700; // 500KB
```
Comment (connorsmith256) [inline host_core/native/hostcore_wasmcloud_native/src/lib.rs:33]: Supernit: 700KB?

File: host_core/native/hostcore_wasmcloud_native/src/lib.rs [hunk 2]
```diff
@@ -104,6 +118,29 @@ rustler::init!(
...
+    let jsoptions = if let Some(d) = config.get("js_domain").cloned() {
+        JetStreamOptions::new().domain(&d)
+    } else {
+        JetStreamOptions::new()
+    };
+    let js = nats::jetstream::JetStream::new(nc, jsoptions);
+    let lattice = config
+        .get("lattice")
+        .cloned()
+        .unwrap_or("default".to_string());
```
Comment (connorsmith256) [inline host_core/native/hostcore_wasmcloud_native/src/lib.rs:136]: Does falling back actually help us recover here? Would it be better to return an error if there's no lattice prefix?

**Review by connorsmith256 — APPROVED** [review]

File: host_core/native/hostcore_wasmcloud_native/src/lib.rs (hunk 1 above)
Comment (autodidaddict) [reply host_core/native/hostcore_wasmcloud_native/src/lib.rs:33]: image

File: host_core/native/hostcore_wasmcloud_native/src/lib.rs (hunk 2 above)
Comment (autodidaddict) [reply host_core/native/hostcore_wasmcloud_native/src/lib.rs:136]: Nearly everything in the wasmCloud ecosystem uses `default` as the lattice when one isn't supplied, so it felt natural to make this behave that way as well.

**Review by connorsmith256 — APPROVED** [review]

File: host_core/lib/host_core/host.ex [hunk 3]
```diff
@@ -371,14 +371,15 @@ defmodule HostCore.Host do
...
       case :ets.lookup(:config_table, :config) do
         [config: config_map] ->
           {"#{config_map[:prov_rpc_host]}:#{config_map[:prov_rpc_port]}",
-           config_map[:prov_rpc_jwt], config_map[:prov_rpc_seed], config_map[:prov_rpc_tls]}
+           config_map[:prov_rpc_jwt], config_map[:prov_rpc_seed], config_map[:prov_rpc_tls],
+           config_map[:rpc_timeout]}
 
         _ ->
-          {"127.0.0.1:4222", "", ""}
+          {"127.0.0.1:4222", "", "", 2000}
```
Comment (brooksmtownsend) [inline host_core/lib/host_core/host.ex:382]: aha so this is why my rpc timeout change didn't configure anything :P

File: host_core/lib/host_core/web_assembly/imports.ex [hunk 4]
```diff
@@ -409,13 +424,31 @@ defmodule HostCore.WebAssembly.Imports do
...
+    if Map.get(ir, "content_length", bsize) > bsize do
+      with {:ok, bytes} <- HostCore.WasmCloud.Native.dechunk_inv(invid) do
+        bytes
+      else
+        _ ->
+          ir["msg"]
+      end
+    else
+      ir["msg"]
+    end
```
Comment (brooksmtownsend) [inline host_core/lib/host_core/web_assembly/imports.ex:449]: super nit, could push the if statement into with
```
with true <- Map.get(ir, "content_length, bsize) > bsize ,
        {:ok, bytes} <- ... do
           bytes
else
_ -> ir["msg"]
end
```

**Review by brooksmtownsend — CHANGES_REQUESTED** [review]
Only real request is that I'd love to see a test with this working, is there a somewhat simple scenario that we could use to test this?

**Comment by autodidaddict** [comment]
> Only real request is that I'd love to see a test with this working, is there a somewhat simple scenario that we could use to test this?

Yeah the tick is finding an actor I can use that has a contract that allows for a payload big enough to chunk

**Review by brooksmtownsend — APPROVED** [review]

File: host_core/mix.exs [hunk 5]
```diff
@@ -33,15 +33,16 @@ defmodule HostCore.MixProject do
       {:wasmex, "~> 0.5.0"},
       {:msgpax, "~> 2.3"},
       {:rustler, "~> 0.22.0"},
+      {:jason, "~> 1.2.2"},
       {:gnat, "~> 1.2"},
       {:cloudevents, "~> 0.4.0"},
       {:uuid, "~> 1.1"},
-      {:jason, "~> 1.2.2"},
+
```
Comment (brooksmtownsend) [inline host_core/mix.exs:40]: Good for Jason, moving its way up the dependency ladder

Comment (autodidaddict) [reply host_core/mix.exs:40]: Everybody loves an underdog story

**Review by brooksmtownsend — APPROVED** [review]

File: .github/workflows/wasmcloud_host.yml [hunk 6]
```diff
@@ -52,36 +52,36 @@ jobs:
...
+      #   run: mix format --check-formatted
 
-      - name: Run Credo
-        working-directory: ${{env.working-directory}}
-        continue-on-error: true # Don't fail entire action with refactoring opportunities for now
-        run: mix credo --strict
+      # - name: Run Credo
+      #   working-directory: ${{env.working-directory}}
+      #   continue-on-error: true # Don't fail entire action with refactoring opportunities for now
+      #   run: mix credo --strict
```
Comment (brooksmtownsend) [inline .github/workflows/wasmcloud_host.yml:62]: We can keep these sections, just need to skip the `mix test` section

**Review by brooksmtownsend — CHANGES_REQUESTED** [review]

**Review by brooksmtownsend — APPROVED** [review]
This time it'll work ™️

**Pull request merged**